tools/check_for_upgrade.sh

```diff
@@ -29,9 +29,9 @@ fi
 # Cancel upgrade if git is unavailable on the system
 whence git >/dev/null || return 0
 
-if [ -f ~/.zsh-update ]
+if [ -f ${ZSH_CACHE_DIR}.zsh-update ]
```

Comment:
Nope, that's fine: in Unix, repeated `/` characters are (mostly) treated the same as a single one. Mostly to make this use case easier, I think.